### PR TITLE
move yaml switch to the footer of forms and flatten navigation

### PR DESCRIPTION
**Pull request by edlerd**
## Done

- move yaml switch to the footer of forms for instance, profile, network and storage pool create and edit
- flatten form navigations

Fixes WD-14519

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - check yaml switch on instance, profile, network and storage pool creation and editing

### CODE REVIEW

**Comment by piperdeck** [comment]
Looking great!

1. Could we have the YAML text field take up the whole width that the inline notification currently takes?

2. Also, would it be possible to make that inline notification dismissable, and use local storage to keep track of whether the user has dismissed it? It's a little unfortunate to have it permanently take up so much space

3. I got some feedback from another designer who briefly worked on a similar feature, and he got some copy approved for the confirmation modal that you see when exiting the YAML editor with unsaved changes:

image

Could we swap this out for our existing warning modal?

**Comment by edlerd** [comment]
Thanks for the design feedback @piperdeck . I addressed the three points, so please give it another round of review.

**Comment by mas-who** [comment]
@piperdeck @edlerd this is a bit of a design question. I think the yaml editor should align with the toggle button on the left and the edit button on the right (soon this will be the save button instead). wdyt?

Screenshot from 2024-08-28 10-46-47

**Comment by mas-who** [comment]
For the network and storage create/edit yaml form, the alignment seems to be different to that of instance and profile sections.

Screenshot from 2024-08-28 10-55-55

**Comment by mas-who** [comment]
I think we should also flatten the navigation for the custom volume config page.

Screenshot from 2024-08-28 11-00-48

**Review by mas-who — COMMENTED** [review]
The change is surprisingly clean, love the refactor of yaml related components and removal of the "Advanced" form menu button. Just one comment regarding the file location for `YamlSwitch`.

File: src/pages/instances/YamlSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { FormikProps } from "formik/dist/types";
+
+interface Props {
+  section?: string;
+  setSection: (section: string) => void;
+  formik: unknown;
+  disableReason?: string;
+}
+
+const YamlSwitch: FC<Props> = ({
```
[the remark this answers is not in the document]
Comment (edlerd) [reply src/pages/instances/YamlSwitch.tsx]: yeah, this is right. moved it over

**Comment by mas-who** [comment]
LGTM code and QA, thanks for the updates :+1:

**Pull request merged**